migration-v4.md

```diff
@@ -0,0 +1,363 @@
+# Migration guide
+
+This document serves as a migration guide for `webpack-dev-server@4.0.0`.
+
+## `after` and `before`
+
+Provides the ability to execute custom middleware after/prior to all other middleware internally within the server.
+
+`before` and `after` were removed in favor `onBeforeSetupMiddleware` and `onAfterSetupMiddleware`:
+
+### webpack-dev-server v3:
+
+```js
+module.exports = {
+  devServer: {
+    after: function (app, server, compiler) {
+      // do fancy stuff
+    },
+    before: function (app, server, compiler) {
+      app.get('/some/path', function (req, res) {
+        res.json({ custom: 'response' });
+      });
+    },
+  },
+};
+```
+
+### webpack-dev-server v4:
+
+```js
+module.exports = {
+  devServer: {
+    onAfterSetupMiddleware: function (app, server, compiler) {
+      // do fancy stuff
+    },
+    onBeforeSetupMiddleware: function (app, server, compiler) {
+      app.get('/some/path', function (req, res) {
+        res.json({ custom: 'response' });
+      });
+    },
+  },
+};
+```
+
+## `client` options
+
+The `sockHost`, `sockPath`, `sockPort`, `clientLogLevel`, `injectClient`, `injectHot`, `overlay`, and `progress` options were moved to `client` option:
```

Comment:
Let's describe here, `sockHost`, `sockPath` and `sockPort` was merged into `webSocketURL`